diff --git a/dist/esm/tempo/Connectors.js b/dist/esm/tempo/Connectors.js
index 6c0d33a650946707b8aea6926e7d7e1f979cc6d4..9608049a950689370919c183e77fe0c65c2aca83 100644
--- a/dist/esm/tempo/Connectors.js
+++ b/dist/esm/tempo/Connectors.js
@@ -6,6 +6,15 @@ import { generatePrivateKey, privateKeyToAccount, } from 'viem/accounts';
 import { Account, WebAuthnP256, WebCryptoP256, walletNamespaceCompat, } from 'viem/tempo';
 import { createConnector } from '../connectors/createConnector.js';
 import { ChainNotConfiguredError } from '../errors/config.js';
+function getKey() {
+    let privateKey = globalThis.localStorage?.getItem('THIS_IS_A_DEMO_PRIVATE_KEY');
+    if (!privateKey) {
+        privateKey = generatePrivateKey();
+        globalThis.localStorage?.setItem('THIS_IS_A_DEMO_PRIVATE_KEY', privateKey);
+    }
+    const account = privateKeyToAccount(privateKey);
+    return { privateKey, publicKey: PublicKey.fromHex(account.publicKey) };
+}
 webAuthn.type = 'webAuthn';
 /**
  * Connector for a WebAuthn EOA.
@@ -95,7 +104,7 @@ export function webAuthn(options) {
                             return undefined;
                         if (!accessKeyOptions)
                             return undefined;
-                        return await WebCryptoP256.createKeyPair();
+                        return getKey();
                     })();
                     return { credential, keyPair, signature: undefined };
                 }
@@ -137,7 +146,7 @@ export function webAuthn(options) {
                             return undefined;
                         if (!accessKeyOptions)
                             return undefined;
-                        return await WebCryptoP256.createKeyPair();
+                        return getKey();
                     })();
                     // If we are provisioning an access key, we will need to sign a key authorization.
                     // We will need the hash (digest) to sign, and the address of the access key to construct the key authorization.
@@ -151,7 +160,7 @@ export function webAuthn(options) {
                             address: accessKeyAddress,
                             expiry: accessKeyOptions?.expiry,
                             strict: accessKeyOptions?.strict ?? false,
-                            type: 'p256',
+                            type: 'secp256k1',
                         });
                         const hash = KeyAuthorization.getSignPayload(keyAuthorization_unsigned);
                         return { keyAuthorization_unsigned, hash };
@@ -205,7 +214,7 @@ export function webAuthn(options) {
                 signature = signedHash;
             }
             else if (keyPair) {
-                accessKey = Account.fromWebCryptoP256(keyPair, {
+                accessKey = Account.fromSecp256k1(keyPair.privateKey, {
                     access: account,
                 });
                 // If we are reconnecting, check if the access key is expired.
diff --git a/src/tempo/Connectors.ts b/src/tempo/Connectors.ts
index 7e8d5a410cb0e7914f0d190c16e259c9717f260b..328431e19a2ed04fd9a6f93ca967a69307063355 100644
--- a/src/tempo/Connectors.ts
+++ b/src/tempo/Connectors.ts
@@ -30,6 +30,16 @@ export type WebAuthnParameters = webAuthn.Parameters
 
 webAuthn.type = 'webAuthn' as const
 
+function getKey() {
+  let privateKey = globalThis.localStorage?.getItem('THIS_IS_A_DEMO_PRIVATE_KEY') as Hex.Hex | null
+  if (!privateKey) {
+    privateKey = generatePrivateKey()
+    globalThis.localStorage?.setItem('THIS_IS_A_DEMO_PRIVATE_KEY', privateKey)
+  }
+  const account = privateKeyToAccount(privateKey)
+  return { privateKey, publicKey: PublicKey.fromHex(account.publicKey) }
+}
+
 /**
  * Connector for a WebAuthn EOA.
  */
@@ -190,7 +200,7 @@ export function webAuthn(options: webAuthn.Parameters) {
           const keyPair = await (async () => {
             if (signHash) return undefined
             if (!accessKeyOptions) return undefined
-            return await WebCryptoP256.createKeyPair()
+            return getKey()
           })()
 
           return { credential, keyPair, signature: undefined }
@@ -241,7 +251,7 @@ export function webAuthn(options: webAuthn.Parameters) {
           const keyPair = await (async () => {
             if (signHash) return undefined
             if (!accessKeyOptions) return undefined
-            return await WebCryptoP256.createKeyPair()
+            return getKey()
           })()
 
           // If we are provisioning an access key, we will need to sign a key authorization.
@@ -258,7 +268,7 @@ export function webAuthn(options: webAuthn.Parameters) {
               address: accessKeyAddress,
               expiry: accessKeyOptions?.expiry,
               strict: accessKeyOptions?.strict ?? false,
-              type: 'p256',
+              type: 'secp256k1',
             })
             const hash = KeyAuthorization.getSignPayload(
               keyAuthorization_unsigned,
@@ -326,7 +336,7 @@ export function webAuthn(options: webAuthn.Parameters) {
       } else if (signedHash) {
         signature = signedHash
       } else if (keyPair) {
-        accessKey = Account.fromWebCryptoP256(keyPair, {
+        accessKey = Account.fromSecp256k1(keyPair.privateKey, {
           access: account,
         })
 
