diff --git a/dist/internal/vite-plugins.js b/dist/internal/vite-plugins.js
--- a/dist/internal/vite-plugins.js
+++ b/dist/internal/vite-plugins.js
@@ -196,8 +196,8 @@ export function llms(config) {
         if (description)
             llmsTxtContent.push(description, '');
         const nav = [];
-        for (const { title, description, path } of results)
-            nav.push(`- [${title}](${path === '/' ? '/index' : path})${description ? `: ${description}` : ''}`);
+        for (const { title: t, description: d, path: p } of results)
+            nav.push(`- [${t}](${p || '/'})${d ? `: ${d}` : ''}`);
         const sitemap = ['<!--', 'Sitemap:', ...nav, '-->', '', ''].join('\n');
         for (const result of results)
             result.content = sitemap + result.content;
diff --git a/dist/internal/config.js b/dist/internal/config.js
--- a/dist/internal/config.js
+++ b/dist/internal/config.js
@@ -84,7 +84,7 @@ export function define(config = {}) {
 }
 export function getConfigFile(options = {}) {
     const { rootDir = process.cwd() } = options;
-    return fs.globSync('vocs.config.{ts,js,mjs,mts}', { cwd: rootDir })[0];
+    if (typeof globalThis.caches !== "undefined") return undefined; return fs.globSync('vocs.config.{ts,js,mjs,mts}', { cwd: rootDir })[0];
 }
 export async function resolve(options = {}) {
     const { server, rootDir = process.cwd() } = options;
diff --git a/dist/internal/langs.js b/dist/internal/langs.js
--- a/dist/internal/langs.js
+++ b/dist/internal/langs.js
@@ -28,7 +28,7 @@ export function infer(options) {
     const langs = new Set(defaultLangs);
     const codeBlockRegex = /```(\w+)/g;
     try {
-        const files = fs.globSync('**/*.{md,mdx}', { cwd });
+        if (typeof globalThis.caches !== "undefined") return langs; const files = fs.globSync('**/*.{md,mdx}', { cwd });
         for (const file of files) {
             const content = fs.readFileSync(path.join(cwd, file), 'utf-8');
             let match;
